tests/samplers_tests/test_nsgaii.py

```diff
@@ -559,6 +573,151 @@ def _create_frozen_trial(
     return trial
 
 
+def test_child_generation_strategy_invalid_value() -> None:
+    with pytest.raises(ValueError):
+        NSGAIIChildGenerationStrategy(
+            mutation_prob=1.2,
+            crossover=UniformCrossover(),
+            crossover_prob=0.9,
+            swapping_prob=0.5,
+        )
+    with pytest.raises(ValueError):
+        NSGAIIChildGenerationStrategy(
+            mutation_prob=-0.2,
+            crossover=UniformCrossover(),
+            crossover_prob=0.9,
+            swapping_prob=0.5,
+        )
+    with pytest.raises(ValueError):
+        NSGAIIChildGenerationStrategy(
+            crossover_prob=1.2,
+            crossover=UniformCrossover(),
+            swapping_prob=0.5,
+        )
+    with pytest.raises(ValueError):
+        NSGAIIChildGenerationStrategy(
+            crossover_prob=-0.2,
+            crossover=UniformCrossover(),
+            swapping_prob=0.5,
+        )
+    with pytest.raises(ValueError):
+        NSGAIIChildGenerationStrategy(
+            crossover_prob=0.9,
+            crossover=UniformCrossover(),
+            swapping_prob=1.2,
+        )
+    with pytest.raises(ValueError):
+        NSGAIIChildGenerationStrategy(
+            crossover_prob=0.9,
+            crossover=UniformCrossover(),
+            swapping_prob=-0.2,
+        )
+    with pytest.raises(ValueError):
+        NSGAIIChildGenerationStrategy(
+            crossover_prob=0.9,
+            crossover=3,  # type: ignore[arg-type]
+            swapping_prob=0.5,
+        )
+
+
+@pytest.mark.parametrize(
+    "mutation_prob,child_params",
+    [(0.0, {"x": 1.0, "y": 0.0}), (1.0, {})],
+)
+def test_child_generation_strategy_mutation_prob(
+    mutation_prob: int, child_params: dict[str, float]
+) -> None:
+    child_generation_strategy = NSGAIIChildGenerationStrategy(
+        crossover_prob=0.0,
+        crossover=UniformCrossover(),
+        mutation_prob=mutation_prob,
+        swapping_prob=0.5,
+        seed=1,
+    )
+    study = MagicMock(spec=optuna.study.Study)
+    search_space = MagicMock(spec=dict)
+    search_space.keys.return_value = ["x", "y"]
+    parent_population = [
+        optuna.trial.create_trial(
+            params={"x": 1.0, "y": 0},
+            distributions={
+                "x": FloatDistribution(0, 10),
+                "y": CategoricalDistribution([-1, 0, 1]),
+            },
+            value=5.0,
+        )
+    ]
+    assert child_generation_strategy(study, search_space, parent_population) == child_params
+
+
+def test_sample_relative() -> None:
```

Comment:
How does this test function correspond to the one in sampler_tests/test_samplers.py? Is it intended to check if generation is correctly set? If so, checking the values of parameters should be skipped, and the test function's name should be made more descriptive.

Reply:
I renamed the test and removed the check for parameters and values